app/js/components/identifiers/addIdentifier.jsx

```diff
@@ -0,0 +1,231 @@
+        this.handleCreateChange = this.handleCreateChange.bind(this);
+        this.callCancelPromise = this.callCancelPromise.bind(this);
+    }
+
+    callNew() {
+        this.setState({ newState: true });
+    }
+
+    handleCreateChange(e) {
+        const label = e.target.name;
+        const value = e.target.value;
+        const { addIdentifiers } = this.state;
+
+        if (label === 'preferred') {
+            addIdentifiers.preferred = e.target.checked;
+        } else {
+            addIdentifiers[label] = value;
+        }
+        this.setState(Object.assign({}, this.state.addIdentifiers, {
+            addIdentifiers,
+        }));
+    }
+
+    callCreate() {
+        const keys = [];
+        const values = [];
+        const { addIdentifiers } = this.state;
+        Object.keys(addIdentifiers).filter((key) => {
+            if (addIdentifiers[key] !== '') {
+                keys.push(key);
+                values.push(addIdentifiers[key]);
+                return addIdentifiers[key];
+            }
+        });
+        const newAddIdentifiers = values.reduce((acc, cur, i) => (acc[keys[i]] = cur, acc), {});
+        if (this.state.addIdentifiers.preferred === true) {
+            apiCall(newAddIdentifiers,
+                'post', `patient/${this.props.stateData.patientUuid}/identifier`)
+                .then((response) => {
+                    apiCall({ preferred: false },
+                        'post',
+                        `patient/${this.props.stateData.patientUuid}/identifier/${this.props.stateData.prefferedIdentifierUuid}`,
+                    )
+                        .then((response) => {
+                            if (response.error) {
+                                this.setState({
+                                    addErrors: {
+                                        error: response.error.message,
+                                    },
+                                });
+                            } else {
+                                this.setState({ newState: false, addIdentifiers: {} });
+                                this.props.fetchIdentifiers();
+                            }
+                        })
+                        .catch(error => toastr.error(error));
+                })
+                .catch(error => toastr.error(error));
+        } else {
+            apiCall(newAddIdentifiers, 'post',
+                `patient/${this.props.stateData.patientUuid}/identifier`)
+                .then((response) => {
+                    if (response.error) {
+                        this.setState({ addErrors: { error: response.error.message } });
+                    } else {
+                        this.setState({ newState: false, addIdentifiers: {} });
+                        this.props.fetchIdentifiers();
+                    }
+                })
+                .catch(error => toastr.error(error));
+        }
+    }
+
+    callCancelPromise() {
```

Comment:
Please use a more descriptive name for this method

Reply:
done